src/udp.rs

```diff
@@ -39,103 +114,293 @@ impl UdpSocket {
         bus: &mut SpiBus,
         remote: SocketAddrV4,
     ) -> Result<(), UdpSocketError<SpiBus::Error>> {
-        self.socket.set_destination_ip(bus, *remote.ip())?;
-        self.socket.set_destination_port(bus, remote.port())?;
+        // We set this variable to true when:
+        /// - We have the same Socket address already set
+        /// - We don't have any previous destination set
+        let is_same = self
+            .destination
+            .filter(|set_destination| set_destination == &remote)
+            .is_some();
+
+        // either no previous destination is set or a new one
+        if !is_same {
```

Comment:
```suggestion
        // either no previous destination is set or a new one has been specified
        if self.destination.map(|dest| dest != &remote).unwrap_or(true) {
```

Reply:
It's a bit less obvious with map and unwrap_or, but we can commit it like that.

Reply:
A `filter()` call seemed far less obvious to me personally. `map()` is a common operation to do on `Option`al data. The unwrap is indeed unintuitive and I'm open to a better idea, just wanted to simplify and make it more logical